contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+import {SupportsNonCompliantERC20} from "../libs/SupportsNonCompliantERC20.sol";
+
+/**
+ * @title OTC Wrapper
+ * @author Ribbon Team
+ * @notice Contract that overlays Gamma Protocol for the OTC related interactions
+ */
+contract OTCWrapper is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeable, ERC2771ContextUpgradeable {
+    using SafeERC20 for IERC20;
+    using SupportsNonCompliantERC20 for IERC20;
+    using SafeMath for uint256;
+
+    AddressBookWrapperInterface public addressbook;
+    MarginRequirementsWrapperInterface public marginRequirements;
+    ControllerWrapperInterface public controller;
+    OracleWrapperInterface public oracle;
+    WhitelistWrapperInterface public whitelist;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    /// @notice emits an event when an order is placed
+    event OrderPlaced(
+        uint256 indexed orderID,
+        address indexed underlyingAsset,
+        bool isPut,
+        uint256 strikePrice,
+        uint256 expiry,
+        uint256 premium,
+        uint256 notional,
+        address indexed buyer
+    );
+
+    /// @notice emits an event when an order is canceled
+    event OrderCancelled(uint256 orderID);
+
+    /// @notice emits an event when an order is executed
+    event OrderExecuted(
+        uint256 indexed orderID,
+        address collateralAsset,
+        uint256 premium,
+        address indexed seller,
+        uint256 indexed vaultID,
+        address oToken,
+        uint256 initialMargin
+    );
+
+    /// @notice emits an event when collateral is deposited
+    event CollateralDeposited(uint256 indexed orderID, uint256 amount);
+
+    /// @notice emits an event when collateral is withdrawn
+    event CollateralWithdrawn(uint256 indexed orderID, uint256 amount);
+
+    /// @notice emits an event when a vault is settled
+    event VaultSettled(uint256 indexed orderID);
+
+    /************************************************
+     *  STORAGE
+     ***********************************************/
+
+    ///@notice order counter
+    uint256 public latestOrder;
+
+    ///@notice fill deadline duration
+    uint256 public fillDeadline;
+
+    // usdc address
+    address public USDC;
+
+    ///@notice address that will receive the product fees
+    address public beneficiary;
+
+    // order status
+    enum OrderStatus {
+        Pending,
+        Succeeded,
+        Failed
+    }
+
+    // struct defining order details
+    struct Order {
+        // underlying asset address
+        address underlying;
+        // collateral asset address
+        address collateral;
+        // option type the vault is selling
+        bool isPut;
+        // option strike price
+        uint256 strikePrice;
+        // option expiry timestamp
+        uint256 expiry;
+        // order premium amount
+        uint256 premium;
+        // order notional
+        uint256 notional;
```

Comment:
if we are going to choose between `notional` and `size`, lets keep size and recalculate notional using oracle price

Reply:
discontinued